File: bindings/julia/test/test_database_run_model.jl
module TestDatabaseRunModel

using Quiver
using Test

include("fixture.jl")

@testset "Run model" begin
    # The happy path (a real exit code from CarbSteeler) is not testable hermetically because
    # the model directory is a hardcoded constant in the native layer; it is verified manually.
    # The in-memory precondition fires before the model directory is consulted, so it is
    # deterministic on every machine/CI runner.
    @testset "Throws on in-memory database" begin
        path_schema = joinpath(tests_path(), "schemas", "valid", "collections.sql")
        db = Quiver.from_schema(":memory:", path_schema)
        @test_throws Quiver.DatabaseException Quiver.run_model!(db)
        Quiver.close!(db)
    end
end

end

File: bindings/python/tests/test_database_run_model.py
from __future__ import annotations

from pathlib import Path

import pytest

from quiverdb import Database, QuiverError


class TestRunModel:
    """Tests for run_model().

    The happy path (a real exit code from CarbSteeler) is not testable hermetically because
    the model directory is a hardcoded constant in the native layer; it is verified manually.
    The in-memory precondition fires before the model directory is consulted, so it is
    deterministic on every machine/CI runner.
    """

    def test_run_model_raises_on_memory_database(self, collections_schema_path: Path) -> None:
        db = Database.from_schema(":memory:", str(collections_schema_path))
        try:
            with pytest.raises(QuiverError, match="in-memory database"):
                db.run_model()
        finally:
            db.close()
